### PR TITLE
[MINOR] Typo fixes

### PR DESCRIPTION
**Pull request by jaceklaskowski**
## What changes were proposed in this pull request?

Typo fixes. No functional changes.
## How was this patch tested?

Built the sources and ran with samples.

### CODE REVIEW

File: streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala [hunk 1]
```diff
@@ -132,7 +132,7 @@ class StreamingContext private[streaming] (
       "both SparkContext and checkpoint as null")
   }
 
-  private[streaming] val isCheckpointPresent = (_cp != null)
+  private[streaming] val isCheckpointPresent = _cp != null
```
Comment (thunterdb) [inline streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala]: nit: since it is package-private, it should be typed: `: Bool = `

File: streaming/src/main/scala/org/apache/spark/streaming/dstream/DStream.scala [hunk 2]
```diff
@@ -269,7 +269,7 @@ abstract class DStream[T: ClassTag] (
       checkpointDuration == null || rememberDuration > checkpointDuration,
       s"The remember duration for ${this.getClass.getSimpleName} has been set to " +
         s" $rememberDuration which is not more than the checkpoint interval" +
-        s" ($checkpointDuration). Please set it to higher than $checkpointDuration."
+        s" ($checkpointDuration). Please set it to a higher value than $checkpointDuration."
```
Comment (thunterdb) [inline streaming/src/main/scala/org/apache/spark/streaming/dstream/DStream.scala]: `a value higher than ..`

File: streaming/src/main/scala/org/apache/spark/streaming/dstream/DStream.scala [hunk 3]
```diff
@@ -277,7 +277,7 @@ abstract class DStream[T: ClassTag] (
     logInfo(s"Slide time = $slideDuration")
     logInfo(s"Storage level = ${storageLevel.description}")
     logInfo(s"Checkpoint interval = $checkpointDuration")
-    logInfo(s"Remember duration = $rememberDuration")
+    logInfo(s"Remember interval = $rememberDuration")
```
Comment (thunterdb) [inline streaming/src/main/scala/org/apache/spark/streaming/dstream/DStream.scala]: why this change?

File: streaming/src/main/scala/org/apache/spark/streaming/dstream/DStreamCheckpointData.scala [hunk 4]
```diff
@@ -45,7 +45,7 @@ class DStreamCheckpointData[T: ClassTag] (dstream: DStream[T])
   /**
    * Updates the checkpoint data of the DStream. This gets called every time
    * the graph checkpoint is initiated. Default implementation records the
-   * checkpoint files to which the generate RDDs of the DStream has been saved.
+   * checkpoint files to which the generated RDDs of the DStream have been saved.
```
Comment (thunterdb) [inline streaming/src/main/scala/org/apache/spark/streaming/dstream/DStreamCheckpointData.scala]: should be `records ... at which ...`

File: streaming/src/main/scala/org/apache/spark/streaming/dstream/DStreamCheckpointData.scala [hunk 5]
```diff
@@ -102,8 +102,8 @@ class DStreamCheckpointData[T: ClassTag] (dstream: DStream[T])
   }
 
   /**
-   * Restore the checkpoint data. This gets called once when the DStream graph
-   * (along with its DStreams) are being restored from a graph checkpoint file.
+   * Restore the checkpoint data. This gets called once while the DStream graph
```
Comment (thunterdb) [inline streaming/src/main/scala/org/apache/spark/streaming/dstream/DStreamCheckpointData.scala]: I am not sure if this comment is accurate, someone from the streaming team should check it. cc @tdas

**Comment by thunterdb** [comment]
I had some small comments from my side. The changes in `ml` look good. Someone from streaming should take over for the rest.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/LogicalPlan.scala [hunk 6]
```diff
@@ -79,13 +79,13 @@ abstract class LogicalPlan extends QueryPlan[LogicalPlan] with Logging {
...
    * Computes [[Statistics]] for this plan. The default implementation assumes the output
-   * cardinality is the product of of all child plan's cardinality, i.e. applies in the case
+   * cardinality is the product of all child plan's cardinality, i.e. applies in the case
    * of cartesian joins.
    *
    * [[LeafNode]]s must override this.
    */
   def statistics: Statistics = {
-    if (children.size == 0) {
+    if (children.isEmpty) {
```
Comment (srowen) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/LogicalPlan.scala]: (Although it's not a typo I am OK with this kind of change if touching the code anyway)

File: sql/core/src/main/scala/org/apache/spark/sql/execution/joins/BroadcastHashJoin.scala [hunk 7]
```diff
@@ -32,7 +32,7 @@ import org.apache.spark.util.collection.CompactBuffer
 /**
  * Performs an inner hash join of two child relations.  When the output RDD of this operator is
  * being constructed, a Spark job is asynchronously started to calculate the values for the
- * broadcasted relation.  This data is then placed in a Spark broadcast variable.  The streamed
+ * broadcast relation.  This data is then placed in a Spark broadcast variable.  The streamed
```
Comment (srowen) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/joins/BroadcastHashJoin.scala]: I think 'broadcasted' is OK as the adjective but 'broadcast' is also used this way so OK

File: streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala (hunk 1 above)
Comment (srowen) [inline streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala]: `: Boolean = `

File: streaming/src/main/scala/org/apache/spark/streaming/dstream/DStreamCheckpointData.scala (hunk 5 above)
Comment (srowen) [inline streaming/src/main/scala/org/apache/spark/streaming/dstream/DStreamCheckpointData.scala]: I'd leave "when" in this case since it is _called_ at the time the graph is restored, not sometime during that process

**Comment by srowen** [comment]
Yeah this is good

File: streaming/src/main/scala/org/apache/spark/streaming/dstream/DStream.scala (hunk 3 above)
Comment (jaceklaskowski) [inline streaming/src/main/scala/org/apache/spark/streaming/dstream/DStream.scala]: It's consistent with "Checkpoint interval" above and the general impression that it's described as "interval" while called "Duration" in the code (due to the type).

**Comment by jaceklaskowski** [comment]
Thanks @srowen @thunterdb for review!

File: streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala [hunk 8]
```diff
@@ -246,9 +246,7 @@ class StreamingContext private[streaming] (
     checkpointDir != null
   }
 
-  private[streaming] def initialCheckpoint: Checkpoint = {
-    if (isCheckpointPresent) _cp else null
-  }
+  private[streaming] def initialCheckpoint: Checkpoint = _cp
```
Comment (srowen) [inline streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala]: This changes the behavior. It could be fine but I don't know if it belongs with typo changes

Comment (srowen) [inline streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala]: @jaceklaskowski can you revert this?

Comment (srowen) [inline streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala]: @jaceklaskowski I'll merge this if you'll fix this part.

Comment (jaceklaskowski) [inline streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala]: Sorry @srowen to keep you waiting. As to the line, look at `isCheckpointPresent` and see that it does what the `if...else` block does, i.e. checks whether `_cp` is `null` and if it is, it returns `false` so `null`. It's a false check here and hence an easy fix (and that's why I proposed it with the other changes).

I can revert it and report another change if you want. Let me know.

Comment (srowen) [inline streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala]: Yes it's functionally equivalent. The method isn't overridden. I suppose that it's conceivable the logic of `isCheckpointPresent` could change later though. Other methods have a similar check too. Although it's a no-op change and small simplification, it doesn't matter in practice to check "if (a != null) a else null" Since the purpose here is just typo fixes, I'd stick to that and not bother with this change.

Comment (srowen) [inline streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala]: @jaceklaskowski let's wrap this up or close it. Undo this change so that this is just about typos and it can be merged

**Comment by srowen** [comment]
Jenkins retest this please

**Comment by srowen** [comment]
Merged to master
